db/memtable_list.h

```diff
@@ -312,7 +312,18 @@ class MemTableList {
   // non-empty (regardless of the min_write_buffer_number_to_merge
   // parameter). This flush request will persist until the next time
   // PickMemtablesToFlush() is called.
-  void FlushRequested() { flush_requested_ = true; }
+  void FlushRequested() {
+    flush_requested_ = true;
+    // If there are some memtables stored in imm() that dont trigger
+    // flush (eg: mempurge output memtable), then update imm_flush_needed.
+    // Note: if race condition and imm_flush_needed is set to true
+    // when there is num_flush_not_started_==0, then there is no
+    // impact whatsoever. Imm_flush_needed is only used in an assert
+    // in IsFlushPending().
+    if (num_flush_not_started_ > 0) {
+      imm_flush_needed.store(true, std::memory_order_release);
```

Comment:
Useful to know - thanks. Note that `imm_flush_needed` isn't an `atomic<bool>` variable introduced in this PR - it has been part of the`MemTableList` struct for a long time, but surprisingly is only used for the `assert` mentioned in the comments here. I thought it would be a useful comment to add in the code because I actually asked myself the same question ("what was this `bool imm_flush_needed` used for").